lib/broccoli/ember-app.js

```diff
@@ -1394,7 +1394,15 @@ EmberApp.prototype.testFiles = function(coreTestTree) {
     overwrite: true,
     annotation: 'TreeMerger (addon-test-support)'
   });
-  var transpiledAddonTestSupportTree = new Babel(addonTestSupportTree, this._prunedBabelOptions());
+
+  var transpiledAddonTestSupportTree
+  if (this._addonInstalled('ember-cli-babel')) {
```

Comment:
The issue is that the transpilation either needs to happen only in `Addon#treeForAddonTestSupport` for all addons _or_ once only in `EmberApp#testFiles`, we can't do both.

If some addons have `ember-cli-babel` installed, then those addons will be double transpiled if we have to run `Babel` in the `EmberApp` again. The approach you outline above would trigger for a scenario where all but one addon is correctly preprocessing, causing everything except that one addon to double transpile.